src/index.ts

```diff
@@ -148,4 +94,38 @@ const chatPlugin: JupyterFrontEndPlugin<void> = {
   }
 };
 
-export default [inlineProviderPlugin, chatPlugin];
+const llmProviderPlugin: JupyterFrontEndPlugin<ILlmProvider> = {
+  id: 'jupyterlab-codestral:llm-provider',
+  autoStart: true,
+  requires: [ICompletionProviderManager, ISettingRegistry],
+  provides: ILlmProvider,
```

Comment:
`ILlmProvider` may read a bit awkward  with the change of casing between the two "L". Maybe `ILLMProvider` or `IAIProvider` could also work? (or something else)